CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,11 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ?
+                previousName : previousName.replacingOccurrences(of: ".\(filextension)", with: "")
```

Comment:
maybe that's can result in bugs if a `.ext` is many times in the filename / path ?

i think better to use here: [deletingPathExtension() -> URL](https://developer.apple.com/documentation/foundation/url/1780065-deletingpathextension)